dm/syncer/syncer.go

```diff
@@ -1684,7 +1686,9 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 	s.tctx.L().Info("replicate binlog from checkpoint", zap.Stringer("checkpoint", lastLocation))
 
 	if s.streamerController.IsClosed() {
-		s.locations.reset(lastLocation)
+		if ce := s.tctx.L().Check(zap.DebugLevel, "update locations"); ce != nil {
+			s.locations.reset(lastLocation)
```

Comment:
branch(not log itself) depends on log level make it hard to understand, and this is not frequent called, prefer to remove it completely